lib/class-wp-rest-widget-areas-controller.php

```diff
@@ -0,0 +1,378 @@
+	 */
+	public function get_items_permissions_check( $request ) {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_user_cannot_edit',
+				__( 'Sorry, you are not allowed to edit sidebars.', 'gutenberg' )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves all widget areas.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		global $wp_registered_sidebars;
+
+		$data = array();
+
+		foreach ( array_keys( $wp_registered_sidebars ) as $sidebar_id ) {
+			$data[ $sidebar_id ] = $this->get_sidebar_data( $sidebar_id );
+		}
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Checks if a given request has access to read a widget area.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_user_cannot_edit',
+				__( 'Sorry, you are not allowed to edit sidebars.', 'gutenberg' )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Retrieves a specific widget area.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		return rest_ensure_response( $this->get_sidebar_data( $request['id'] ) );
+	}
+
+	/**
+	 * Checks if a given REST request has access to update a widget area.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|bool True if the request has access to update the item, error object otherwise.
+	 */
+	public function update_item_permissions_check( $request ) {
+		if ( ! current_user_can( 'edit_theme_options' ) ) {
+			return new WP_Error(
+				'rest_user_cannot_edit',
+				__( 'Sorry, you are not allowed to edit sidebars.', 'gutenberg' )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Updates a single widget area.
+	 *
+	 * @since 5.7.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$sidebar_id      = $request->get_param( 'id' );
+		$sidebar_content = $request->get_param( 'content' );
+		global $wp_registered_sidebars;
+
+		$sidebars = wp_get_sidebars_widgets();
+		$sidebar  = $sidebars[ $sidebar_id ];
+		$post_id  = wp_insert_post(
+			array(
+				'ID'           => is_numeric( $sidebar ) ? $sidebar : 0,
+				'post_content' => $sidebar_content,
```

Comment:
We need to think through how updates to legacy widgets should be handled. Here, we're not updating the existing widgets that are in the database. This means that we'll lose any updates made if we go back to the old widgets screen. I'm thinking we should make `core/legacy-widget` handle this as this would match how widgets currently work where changes made to a widget are saved independently of changes made to a sidebar.

Reply:
Resolving this as we spoke offline and agreed `core/legacy-widget` should handle this.